samples/csharp_dotnetcore/45.state-management/README.md

```diff
@@ -4,6 +4,41 @@ This sample demonstrates how to save user and conversation data in an ASP.Net Co
 The bot maintains conversation state to track and direct the conversation and ask the user questions.
 The bot maintains user state to track the user's answers.
 
+By default this bot uses MemoryStorage for Conversation and User state.
+Memory Storage is great for testing purposes, but in a production scenario you will need to use persistent storage.
+Startup.cs contains commented examples of using Azure Blob Storage or Cosmos DB instead.
+
+# Azure Blob Storage
+
+To use Azure Blob Storage, create a Blob Storage account in your Azure subscription. You can then use the following code to
+create your storage layer, passing in your Blob Storage connection string and a blob container name.  
+
+***Note: You do not need to create the container manually, the bot will create the container for you if it does not exist.***
+
+```cs
+    var storage = new AzureBlobStorage("<blob-storage-connection-string>", "bot-state");
+```
+
+# CosmosDB Storage
+
+To use CosmosDB Storage, you need to create a CosmosDB instance in your Azure subscription. You can then use the following code to
+create your storage layer.  
+
+***Note: It is your responsibility to create an appropriate database within your CosmosDB instance. However, you should **not**
+create the container within the database, as the bot will do this for you and ensure the container is configured correctly.***
+
+```cs
+    var cosmosDbStorageOptions = new CosmosDbPartitionedStorageOptions()
+    {
+        CosmosDbEndpoint = "<endpoint-for-your-cosmosdb-instance>",
+        AuthKey = "<your-cosmosdb-auth-key>",
+        DatabaseId = "<your-database-id>",
+        ContainerId = "<cosmosdb-container-id>"
```

Comment:
Not saying it should or needs to be this way, but what are your thoughts on hard-coding the strings like this vs. putting them in `appsettings.json` and then bringing them in through either [`ConfigurationBuilder`](https://github.com/microsoft/BotBuilder-Samples/blob/4f978577e358d193c4a63ffdab4d10c2fc4b4874/experimental/multilingual-luis/csharp_dotnetcore/Startup.cs#L33) and/or [`BotServices`](https://github.com/microsoft/BotBuilder-Samples/blob/b49159c1cf3b6bd57cf7edf6d8f70a6d72d5a840/samples/csharp_dotnetcore/14.nlp-with-dispatch/BotServices.cs)?

Reply:
I deliberately didn't use the app settings to keep things as obvious and focused as possible.  I would be interested in getting input from @cleemullins here to get thoughts though.

Reply:
I thought that might be the case and lean towards that direction. On the support side, I have trouble pointing people to the appropriate sample (which I'd prefer to "here's *my* code...") when they want to use appsettings for this kind of thing.

Reply:
Using a code comment like this is in line w/ our existing samples and seems the correct approach.